### PR TITLE
NXDRIVE-2751: Sanitize document title (or name) to remove the dot at the end during the synchronization

### PR DESCRIPTION
**Pull request by mr-shekhar**

### CODE REVIEW

File: nxdrive/client/local/windows.py [hunk 1]
```diff
@@ -36,6 +36,10 @@ class LocalClient(LocalClientMixin):
 
     def change_created_time(self, filepath: Path, d_ctime: datetime, /) -> None:
         """Change the created time of a given file."""
+        log.info(f"Editing {filepath!r}")
+        if filepath != safe_filename(filepath):
+            filepath = safe_filename(filepath)
+            log.info(f"filepath sanitized to {filepath!r}")
```
Comment (rom1win) [inline nxdrive/client/local/windows.py:42]: Remove it

**Review by rom1win — COMMENTED** [review]
Don't forget changelog

File: nxdrive/utils.py [hunk 2]
```diff
@@ -523,7 +523,9 @@ def safe_filename(name: str, /, *, replacement: str = "-") -> str:
     """
     return (
         # Windows doesn't allow whitespace at the end of filenames
```
Comment (rom1win) [inline nxdrive/utils.py:525]:
```suggestion
        # Windows doesn't allow whitespace or dots at the end of filenames
```

File: nxdrive/utils.py [hunk 3]
```diff
@@ -523,7 +523,9 @@ def safe_filename(name: str, /, *, replacement: str = "-") -> str:
...
-        (name.rstrip() if WINDOWS else name)
+        (
+            name.rstrip(" .") if WINDOWS else name
+        )  # Sanitize the doc name to remove dot at the end. See NXDRIVE-2751 for details.
```
Comment (rom1win) [inline nxdrive/utils.py:528]:
```suggestion
        )
```

File: nxdrive/client/local/windows.py (hunk 1 above)
Comment (mr-shekhar) [reply nxdrive/client/local/windows.py]: Done

**Pull request merged**